core/templates/domain/recommendations/exploration-recommendations-backend-api.service.ts

```diff
@@ -30,6 +30,7 @@ import {
 // If we send these params always, the request URL would have something like
 // '?collection_id=null' and the backend would start looking for a collection
 // with id "null" which is not correct.
+// eslint-disable-next-line @typescript-eslint/consistent-type-definitions
 type RecommendationsUrlParams = {
```

Comment:
In the file "core/templates/domain/recommendations/exploration-recommendations-backend-api.service.ts", I'm facing an issue when trying to change this definition to use the interface keyword.

When I change it to the following:
![image](https://user-images.githubusercontent.com/78612244/139651365-3a6c725f-47bc-40f6-a7f8-233594596c52.png)

and an object of this type is created:
![image](https://user-images.githubusercontent.com/78612244/139653869-0e2375fd-c4de-4efd-a39f-8aa3d267db61.png)


which is passed to a `http.get` method:
![image](https://user-images.githubusercontent.com/78612244/139651550-937d5c07-7369-48aa-ab39-cd82126b1af3.png)

I get this error:
![image](https://user-images.githubusercontent.com/78612244/139651614-8440c33d-8b8b-4bba-848b-146571b285d5.png)

In the method definition for `http.get`, this is the type definition for the value of the parameter `params`:
![image](https://user-images.githubusercontent.com/78612244/139651869-33093b42-1451-42ba-ac17-ad880455e501.png)

meaning the value assigned to the `params` key should be an object in which the key type is `string`, and value type is `string` or `string array`. The definition `RecommendationsUrlParams` satisfies this criteria, since both the keys and the values are strings. However, the get method still doesn't accept the argument.

I then found [this thread](https://stackoverflow.com/questions/37006008/typescript-index-signature-is-missing-in-type) and the first answer states that:

"They are equivalent types at runtime (indeed, they're the exact same value), but a big part of TypeScript's safety comes from the fact that these aren't the same... This is what stops you accidentally reading and writing totally non-existent properties on objects."

So even though both `RecommendationsUrlParams` and the type `[param: string]: string | string[]` are basically similar (key type of `string` and value type of `string`), typescript doesn't agree because the types aren't exactly the same.

I also found another answer in the same thread, further below, which stated that using the type keyword instead of the interface keyword to write the type definition works fine and throws no errors (this is why it worked fine so far, but it throws an error when I change it to use the interface keyword).

### From what I understand, I have two options:

**1. Change the type definition:**

If I change the type definition to this:
![image](https://user-images.githubusercontent.com/78612244/139654277-741f4448-309a-4e77-8eb3-7fbef9947462.png)

Typescript doesn't complain anymore, and it works fine, because it matches the type of the `params` value.
_However, we lose type safety, since now any property can be assigned to an object of this type._

**2. Ignore this as an exception:**

Since using the `interface` keyword is what's causing the problem, I can ignore this as an exception and let it stay in its original form:
![image](https://user-images.githubusercontent.com/78612244/139655036-d5e02c03-bea9-48b1-bed7-e10b9c82efbc.png)

Currently, I am using the second solution to make this draft PR and discuss this issue.
I would like to know which solution to go with, also if there is a better solution that I missed. PTAL.

Reply:
I think the usage of type is fine in that case, just please add an explanatory comment for it in the code.

Reply:
And great analysis btw!

Reply:
Thank you :)

I've added a comment explaining the reason behind the use of a type alias instead of interface.
I've also used semicolons instead of commas, as per the changes you suggested down below.

PTAL @vojtechjelinek